x-pack/plugins/cloud_security_posture/public/pages/compliance_dashboard/compliance_dashboard.tsx

```diff
@@ -305,60 +330,75 @@ export const ComplianceDashboard = () => {
     enabled: isCloudSecurityPostureInstalled,
   });
 
-  useEffect(() => {
-    if (hasUserSelectedTab) {
-      return;
+  const location = useLocation();
+  const history = useHistory();
+  const { services } = useKibana();
+
+  const currentTabUrlState: PosturePolicyTemplate | undefined = useMemo(() => {
+    let tab: PosturePolicyTemplate | undefined;
+    if (location.pathname === cloudPosturePages.kspm_dashboard.path) {
+      tab = POSTURE_TYPE_KSPM;
     }
 
-    const preferredDashboard = getDefaultTab(
-      getSetupStatus,
-      getCspmDashboardData.data,
-      getKspmDashboardData.data
-    );
-    setSelectedTab(preferredDashboard);
-  }, [
-    getCspmDashboardData.data,
-    getCspmDashboardData.data?.stats.totalFindings,
-    getKspmDashboardData.data,
-    getKspmDashboardData.data?.stats.totalFindings,
-    getSetupStatus,
-    getSetupStatus?.cspm?.status,
-    getSetupStatus?.kspm?.status,
-    hasUserSelectedTab,
-  ]);
+    if (location.pathname === cloudPosturePages.cspm_dashboard.path) {
+      tab = POSTURE_TYPE_CSPM;
+    }
 
-  const tabs = useMemo(
-    () =>
-      isCloudSecurityPostureInstalled
-        ? [
-            {
-              label: i18n.translate('xpack.csp.dashboardTabs.cloudTab.tabTitle', {
-                defaultMessage: 'Cloud',
-              }),
-              'data-test-subj': CLOUD_DASHBOARD_TAB,
-              isSelected: selectedTab === POSTURE_TYPE_CSPM,
-              onClick: () => {
-                setSelectedTab(POSTURE_TYPE_CSPM);
-                setHasUserSelectedTab(true);
-              },
-              content: <TabContent posturetype={POSTURE_TYPE_CSPM} />,
+    // if the location is /dashboard or cloudPosturePages.dashboard.path, then return undefined
+    return tab;
+  }, [location.pathname]);
+
+  const preferredTabUrlState = useMemo(
+    () => getDefaultTab(getSetupStatus, getCspmDashboardData.data, getKspmDashboardData.data),
+    [getCspmDashboardData.data, getKspmDashboardData.data, getSetupStatus]
+  );
+
+  const tabs = useMemo(() => {
+    const navigateToPostureTypeDashboardTab = (pathname: string) => {
+      history.push({
+        pathname,
+        search: encodeQuery({
+          // Set query language from user's preference
+          query: services.data.query.queryString.getDefaultQuery(),
+          filters: services.data.query.filterManager.getFilters(),
+        }),
+      });
+    };
+
+    const selectedTab = currentTabUrlState ?? preferredTabUrlState;
+    return isCloudSecurityPostureInstalled
+      ? [
+          {
+            label: i18n.translate('xpack.csp.dashboardTabs.cloudTab.tabTitle', {
+              defaultMessage: 'Cloud',
```

Comment:
I believe since we are using translation, we just preserve the default text message. `Cloud`  and `Kubernetes`. I don't know if it's necessary to add a parameter since we are only using one word.